### PR TITLE
Avoid nil pointer dereference in (*Cluster).createContainer

### PR DESCRIPTION
**Pull request by wrotki**
Swarm Engine.Create can return nil,nil when refresh fails due to a network hiccup. It causes a panic described by issue https://github.com/docker/swarm/issues/1081 .

### CODE REVIEW

**Comment by aluzzardi** [comment]
We should identify the root cause - this is not supposed to happen.

As a better fix, I'd rather have `Engine.Create()` return an error instead of nil container.

As a convention, one can assume that if `err == nil`, then the value is ready to use (no need to check for nil-ness)

**Comment by wrotki** [comment]
Agreed , returning error from EngineCreate is better.

The root cause - no clue how hard it is to reproduce the problem, I accidentally spotted a single panic instance in my logs. From what understand about what sequence might lead to this, it'd need to be a container creation call returning success, plus subsequent query for the newly created container returning zero elements.  Will try to run more tests in attempt to catch docker red-handed.

**Comment by aluzzardi** [comment]
Awesome - thanks @wrotki 

Feel free to update this PR whenever you can (with the error returned from Engine create) and we'll merge it.

**Comment by abronan** [comment]
:+1: encountered this recently, would be cool to see this merged ASAP.

**Comment by wrotki** [comment]
Alright - I'm in the middle of something right now, but I should be able to update the PR within a day

**Comment by vieux** [comment]
ping @aluzzardi

File: cluster/engine.go [hunk 1]
```diff
@@ -425,7 +425,11 @@ func (e *Engine) Create(config *ContainerConfig, name string, pullImage bool) (*
 	e.RLock()
 	defer e.RUnlock()
 
-	return e.containers[id], nil
+	justCreated := e.containers[id]
```
Comment (aluzzardi) [inline cluster/engine.go]: Sorry for being pedantic but why not `container` or `c`? Semantically speaking, everything on the left side of `:=` was *just created* :)

Comment (wrotki) [inline cluster/engine.go]: Fine with me, will update in a sec

**Comment by aluzzardi** [comment]
Thanks @wrotki !

LGTM

**Pull request merged**